src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs

```diff
@@ -4521,55 +4522,67 @@ private bool AddTaskContinuationComplex(object tc, bool addBeforeOthers)
             Debug.Assert(tc != null, "Expected non-null tc object in AddTaskContinuationComplex");
 
             object? oldValue = m_continuationObject;
+            if (oldValue == s_taskCompletionSentinel)
```

Comment:
As long as we're changing this method, let's add an assert that codifies the comment above "// THIS METHOD ASSUMES THAT m_continuationObject IS NOT NULL.".